catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -0,0 +1,163 @@
+
+        /*
+         * Use the uri to look up the namespace prefix and append it and the localName to the start tag
+         */
+        xmlSnippet.append("<").append(namespaceMapping.get(uri)).append(":").append(localName);
+        
+        /*
+         * If the elements namespace hasn't been declared in an "xmlns" attribute earlier in the XML snippet, 
+         * append it to the start tag
+         */
+        if (!namespacesAdded.contains(namespaceMapping.get(uri))) {
+            xmlSnippet.append(" xmlns:").append(namespaceMapping.get(uri)).append("=\"").append(uri)
+                    .append("\"");
+            namespacesAdded.add(namespaceMapping.get(uri));
+        }
+        
+        /*
+         * Loop through the attributes, if any of their namespaces haven't been declared previously 
+         * in the XML snippet, append it to the start tag
+         */
+        for (int i = 0; i < atts.getLength(); i++) {
+            String attUri;
+            if (!(attUri = atts.getURI(i)).isEmpty() && !namespacesAdded
+                    .contains(namespaceMapping.get(attUri))) {
+                xmlSnippet.append(" xmlns:").append(namespaceMapping.get(attUri)).append("=\"")
+                        .append(attUri).append("\"");
+                namespacesAdded.add(namespaceMapping.get(attUri));
+            }
+        }
+        
+        /*
+         * Loop through the attributes and append them, prefixed with the proper namespace
+         * We loop through the attributes twice to ensure all "xmlns" attributes are declared before
+         * other attributes
+         */
+        for (int i = 0; i < atts.getLength(); i++) {
+            xmlSnippet.append(" ");
+            if (!atts.getURI(i).isEmpty()) {
+                xmlSnippet.append(namespaceMapping.get(atts.getURI(i))).append(":");
+            }
+            xmlSnippet.append(atts.getLocalName(i)).append("=\"").append(atts.getValue(i))
+                    .append("\"");
+        }
+        xmlSnippet.append(">");
+        return this;
+    }
+
+    /**
+     * Method used in a {@link SaxEventHandler#startElement(String, String, String, Attributes)}
+     * to reconstruct the end tag of an XML element. (Could potentially be used elsewhere, but one would have to ensure correct use)
+     *
+     * @param uri       the namespaceURI that is passed in by {@link SaxEventHandler}
+     * @param localName the localName that is passed in by {@link SaxEventHandler}
+     * @return this
+     */
+    public SaxEventToXmlElementConverter toElement(String uri, String localName) {
+        /*
+         * Append the properly prefixed end tag to the XML snippet
+         */
+        xmlSnippet.append("</").append(namespaceMapping.get(uri)).append(":").append(localName)
+                .append(">");
+        return this;
+    }
+
+    /**
+     * Method used in a {@link SaxEventHandler#startElement(String, String, String, Attributes)}
```

Comment:
This is a general comment, but I'm not sure about javadoc in the form 'I'm an `x` used by `y` to do `z`'. It feels like an inversion of object-caller flow, even if it's only in the javadocs. I'd prefer wording that avoids referencing callers, instead simply explaining what something does or what it can be used for.